news/7555.removal

```diff
@@ -1,9 +1,2 @@
-Building of local directories is now done in place. Previously pip did copy the
-local directory tree to a temporary location before building. That approach had
-a number of drawbacks, among which performance issues, as well as various
-issues arising when the python project directory depends on its parent
-directory (such as the presence of a VCS directory). The user visible effect of
-this change is that secondary build artifacts, if any, may therefore be created
-in the local directory, whereas before they were created in a temporary copy of
-the directory and then deleted. This notably includes the ``build`` and
-``.egg-info`` directories in the case of the setuptools build backend.
+Building of local directories is now done in place, instead of a temporary
+location containing a copy of the directory tree.
```

Comment:
@pradyunsg I think that calling out the *effect* of this change (namely that `.egg-info` or `build` directories get created) in the changelog could have increased uor chances to get better feedback during the beta period.

Reply:
You are correct! And...

> we can fix everything except the NEWS fragment later

I just realized that this is not true.

RTD is smart enough to detect beta release numbers, so that they don't end up on `stable`. So... we can update this file on `master` directly to add this information.